com.unity.cinemachine/Runtime/Behaviours/CinemachineConfiner2D.cs

```diff
@@ -262,6 +262,10 @@ public void Invalidate()
                 float aspectRatio, out bool confinerStateChanged)
             {
                 confinerStateChanged = false;
+                
+                if (boundingShape2D == null)
```

Comment:
Prior to your change, the old code called Invalidate() if boundingShape2D is null.  Why have you changed that?

Reply:
I reverted it, just to not add a regression accidentally. There may be some cases where invalidate is needed. E.g. user removes boundingShape2D then adds it back but in some modified version. In this case, if we might not re-calculate cache.